drivers/dac/ad3530r/ad3530r.c

```diff
@@ -0,0 +1,1093 @@
+* INCIDENTAL,SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+* * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS
+* OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+* HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+* LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+* OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+* DAMAGE.
+******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "ad3530r.h"
+#include <stdlib.h>
+#include <string.h>
+#include "no_os_delay.h"
+#include "no_os_error.h"
+#include "no_os_print_log.h"
+#include "no_os_gpio.h"
+#include "no_os_spi.h"
+#include "no_os_util.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+#define AD3530R_DATA_INDEX(x)               x ? 1 : 2
+#define AD3530R_INSTR_LEN(x)                AD3530R_DATA_INDEX(x)
+#define AD3530R_ADDR_INDEX(x)               x ? 0 : 1
+#define AD3530R_BUFF_LEN(x)                 x ? 2 : 3
+#define AD3530R_CRC_BUFF_LEN(x)             x ? 3 : 4
+#define AD3530R_CRC_INDEX(x)                AD3530R_ADDR_INDEX(x) + 2
+
+/******************************************************************************/
+/************************ Variable Declarations *******************************/
+/******************************************************************************/
+static struct ad3530r_transfer_config multi_cfg = {
+	.single_instr = 0, // Sets the streaming mode.
+	.stream_length_keep_value = 1 // Prevents the stream length value from
+	// automatically resetting to zero.
+};
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/******************************************************************************/
+
+/**
+ * @brief Updates the interface configuration.
+ * @param desc - The device structure.
+ * @param cfg - Updated interface configuration structure.
+ * @return 0 in case of success, negative error otherwise.
+ */
+int ad3530r_update_interface_cfg(struct ad3530r_desc *desc,
+				 struct ad3530r_transfer_config *cfg)
+{
+	int ret ;
+
+	if (!desc || !cfg)
+		return -EINVAL;
+
+	ret = ad3530r_spi_write_mask(desc,
+				     AD3530R_REG_ADDR_INTERFACE_CONFIG_A,
+				     AD3530R_MASK_ADDR_ASCENSION,
+				     cfg->addr_asc);
+	if (NO_OS_IS_ERR_VALUE(ret))
```

Comment:
We don't really use `NO_OS_IS_ERR_VALUE()` for these cases anymore. If your functions only return negative error codes, use this instead:
```
if (ret)
    return ret;
```